src/main/java/com/exactpro/th2/fix/client/Main.java

```diff
@@ -114,45 +114,53 @@ public static void main(String[] args) throws Exception {
-                Path pathToDictionary = Files.createFile(getPathToDictionary(temporaryDirectory, zipName));
-                Files.write(pathToDictionary, zipInputStream.readAllBytes());
-                new DataDictionary(pathToDictionary.toString()); //check that xml file contains the correct values
+            } catch (IncorrectFixFileNameException | ConfigError e) {
+                throw new Exception("Failed to unzip dictionaries along the path: " + zipEntry.getName(), e);
             }
-        } catch (IOException | ConfigError e) {
+        } catch (IOException e) {
             throw new Exception("Failed to create DataDictionary", e);
         }
 
         for (int i = 0; i < settings.sessionSettings.size(); i++) {
-            String appDataDictionary = settings.sessionSettings.get(i).getAppDataDictionary();
-            String transportDataDictionary = settings.sessionSettings.get(i).getTransportDataDictionary();
-            String dataDictionary = settings.sessionSettings.get(i).getDataDictionary();
-            if (dataDictionary != null) {
-                Path pathToDataDictionary = getPathToDictionary(temporaryDirectory, dataDictionary);
-                settings.sessionSettings.get(i).setDataDictionary(pathToDataDictionary.toString());
-            } else {
+            if (settings.sessionSettings.get(i).getBeginString().equals("FIXT.1.1")) {
+
+                Path appDataDictionary = settings.sessionSettings.get(i).getAppDataDictionary();
+                Path transportDataDictionary = settings.sessionSettings.get(i).getTransportDataDictionary();
+
                 if (transportDataDictionary == null || appDataDictionary == null) {
                     String sessionID = FixBeanUtil.getSessionID(settings.sessionSettings.get(i)).toString();
                     throw new IllegalStateException("Failed to set dictionary path for session " + sessionID);
                 }
                 Path pathToTransportDataDictionary = getPathToDictionary(temporaryDirectory, transportDataDictionary);
                 settings.sessionSettings.get(i).setTransportDataDictionary(pathToTransportDataDictionary.toString());
-
                 Path pathToAppDataDictionary = getPathToDictionary(temporaryDirectory, appDataDictionary);
                 settings.sessionSettings.get(i).setAppDataDictionary(pathToAppDataDictionary.toString());
+            } else {
+                Path dataDictionary = settings.sessionSettings.get(i).getDataDictionary();
+                if (dataDictionary == null) {
+                    String sessionID = FixBeanUtil.getSessionID(settings.sessionSettings.get(i)).toString();
+                    throw new IllegalStateException("Failed to set dictionary path for session " + sessionID);
+                }
+                Path pathToDataDictionary = getPathToDictionary(temporaryDirectory, dataDictionary);
+                settings.sessionSettings.get(i).setDataDictionary(pathToDataDictionary.toString());
```

Comment:
> settings.sessionSettings.get(i)

Did not you get tired of writing it every time you need to access current element?
I've counted 9 occurrences...